diffpy/pdfmorph/pdfmorphapp.py

```diff
@@ -227,9 +239,130 @@ def custom_error(self, msg):
+        parser.custom_error("Too many arguments. Go to --help for usage.")
+
+    # Parse paths
+    morph_file = Path(pargs[0])
+    if not morph_file.is_file():
+        parser.custom_error(f"{morph_file} is not a file. Go to --help for usage.")
+    target_directory = Path(pargs[1])
+    if not target_directory.is_dir():
+        parser.custom_error(f"{target_directory} is not a directory. Go to --help for usage.")
+
+    # Sort files in directory
+    target_list = list(target_directory.iterdir())
+    if opts.temp:
+        # Sort by temperature
+        try:
+            target_list = tools.temperature_sort(target_list)
+        except ValueError:
+            parser.custom_error("All file names in directory must end in _#K.gr or _#K.cgr (where # is a float) to use "
+                                "the --temperature option.")
+    else:
+        # Default is alphabetical sort
+        target_list.sort()
+
+    # Disable single morph plotting and saving
+    plot_opt = opts.plot
+    opts.plot = False
+    save_opt = opts.savefile
+    opts.savefile = None
+
+    # Do not morph morph_file against itself if it is in the same directory
+    if morph_file in target_list:
+        target_list.remove(morph_file)
+
+    # Morph morph_file against all other files in target_directory
+    results = []
+    for target_file in target_list:
+        if target_file.is_file:
+            results.append([
+                target_file.name,
+                single_morph(parser, opts, [morph_file, target_file], stdout_flag=False),
+            ])
+
+    # Input parameters used for every morph
+    inputs = [None, None]
+    inputs[0] = f"# Morphed file: {morph_file.name}"
+    inputs[1] = "\n# Input morphing parameters:"
+    inputs[1] += f"\n# scale = {opts.scale}"
+    inputs[1] += f"\n# stretch = {opts.stretch}"
+    inputs[1] += f"\n# smear = {opts.smear}"
+
+    # If print enabled
+    if stdout_flag:
+        # Separated for saving
+        print(f"\n{inputs[0]}{inputs[1]}")
+
+        # Results from each morph
+        for entry in results:
+            outputs = f"\n# Target: {entry[0]}\n"
+            outputs += "# Optimized morphing parameters:\n"
+            outputs += "\n".join(f"# {i[0]} = {i[1]:.6f}" for i in entry[1])
+            print(outputs)
+
+    rws = []
+    target_labels = []
+    results_length = len(results)
+    for entry in results:
+        if opts.temp:
+            name = entry[0]
+            target_labels.append(tools.extract_temperatures([name])[0])
+        else:
+            target_labels.append(entry[0])
+        for item in entry[1]:
+            if item[0] == "Rw":
+                rws.append(item[1])
 
-    if len(pargs) != 2:
-        parser.error("You must supply FILE1 and FILE2")
+    if save_opt is not None:
+        # Save table of Rw values
+        try:
+            with open(save_opt, 'w') as outfile:
+                # Header
+                print(f"{inputs[0]}\n{inputs[1]}", file=outfile)
```

Comment:
this looks a bit too chatty to me?  Maybe enclose it in a `if verbose:` (collect verbose as  `-v`, `--verbose`).  In general I prefer my code to be less "chatty" because then the outputs just start looking like noise and important messages get lost and overlooked.   using verbose allows debugging when things are going wrong and the user can't figure out why.

Reply:
Should I also change this for single morphs as well? (Line 526 of this file and onward.)

Reply:
yes, everywhere......